lib/pure/ropes.nim

```diff
@@ -148,15 +152,25 @@ proc rope*(f: BiggestFloat): Rope {.rtl, extern: "nro$1BiggestFloat".} =
   ## Converts a float to a rope.
   result = rope($f)
 
+proc cachingNotSupportedError() {.compileTime.} =
+  raise newException(AssertionError,
+      "Rope caching is not supported in static context")
+
 proc enableCache*() {.rtl, extern: "nro$1".} =
   ## Enables the caching of leaves. This reduces the memory footprint at
   ## the cost of runtime efficiency.
-  cacheEnabled = true
+  when nimvm:
```

Comment:
Just make it a no-op at compile-time? Seems overkill to report that as an error.